plugin/discovery/meta.go

```diff
@@ -16,21 +14,14 @@ type PluginMeta struct {
 	Name string
 
 	// Version is the semver version of the plugin, expressed as a string
-	// that might not be semver-valid. (Call VersionObj to attempt to
-	// parse it and thus detect if it is invalid.)
-	Version string
+	// that might not be semver-valid.
+	Version VersionStr
```

Comment:
This was primarily to enable me to move the `Parse` method onto the version itself rather than having it live up on the meta object... previously it was necessary to say `meta.ParseVersion()` to get the parsed version, but I found it useful to be able to parse versions in isolation so now we can say `meta.Version.Parse()`, or indeed `version.Parse()`.

Other than that it's really just a way to express that there's a particular format of string expressed here, which could equally be expressed as a comment (and in fact, already is) so if you think this feels weird I can try to shape it a different way, e.g. via a top-level function instead of a method.